sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -47,6 +35,17 @@ enum
   AZ_HUB_CLIENT_DEFAULT_MQTT_TELEMETRY_QOS = 0
 };
 
+/**
+ * @brief Azure IoT Hub Client twin_content_type option values.
+ *
+ */
+typedef enum
+{
+  AZ_IOT_HUB_CLIENT_OPTION_TWIN_CONTENT_TYPE_UNDEFINED = 0,
```

Comment:
Why undefined? We know there only can be two types. Default to JSON. Also consider making JSON the first one.

Reply:
or AZ_IOT_HUB_CLIENT_OPTION_TWIN_CONTENT_TYPE_UNDEFINED -> AZ_IOT_HUB_CLIENT_OPTION_TWIN_CONTENT_TYPE_DEFAULT

Reply:
Hmm, I was thinking for backwards compat reasons we would need the option for the username to not include the content-type at all, but I think that's not needed now. With the release of this feature, it will be using a defined hub api that can handle content type in the username.  In this case, then yes we can always include JSON as the chosen content type in the username, and I can get rid of undefined.

Reply:
I was thinking not sending it to save bytes across the wire.
AZ_IOT_HUB_CLIENT_OPTION_TWIN_CONTENT_TYPE_DEFAULT_JSON?

Reply:
That's a better reason not to.  Thanks for thinking of that.  
If we want to not be able to send anything, then there needs to be a generic undefined/default so the logic will recognize the option has not been set.  This assumes that if an app does set the option to use JSON, then we SHOULD send it across the wire.

The other approach is to always have the default option set to JSON, and even if an app decides to set it again, we will never send the JSON content type selection across the wire.  "default-content-type:application%2Fjson" is never used.